climada/hazard/tc_tracks.py

```diff
@@ -2867,3 +2868,64 @@ def _zlib_from_dataarray(data_var: xr.DataArray) -> bool:
     if np.issubdtype(data_var.dtype, float) or np.issubdtype(data_var.dtype, int):
         return True
     return False
+
+
+def compute_track_density(
+    tc_track: TCTracks,
+    res: int = 5,
+    time_step: float = 1,
+    density: bool = False,
+    filter_tracks: bool = True,
+) -> tuple[np.ndarray, tuple]:
+    """Compute absolute and normalized tropical cyclone track density. First, the function ensure
+    the same temporal resolution of all tracks by calling :py:meth:`equal_timestep`. Second, it
+    creates 2D bins of the specified resolution (e.g. 1° x 1°). Third, since tracks are not lines
+    but a series of points, it counts the number of points per bin. Lastly, it returns the absolute
+    or normalized count per bin. This function works under the hood of :py:meth:`plot_track_density`
+    but can be used separtly as input data for more sophisticated track density plots.
+    If filter track is True, only a maximum of one point is added to each grid cell for every track.
+    Hence, the resulting density will represent the number of differt tracksper grid cell.
+
+    Parameters:
+    ----------
+    res: int (optional) Default: 5°
+        resolution in degrees of the grid bins in which the density will be computed
+    time_step: float (optional) default: 1h
+        temporal resolution in hours to be apllied to the tracks, to ensure that every track
+        will have the same resolution.
+    density: bool (optional) default: False
+        If False it returns the number of samples in each bin. If True, returns the
+        probability density function at each bin computed as count_bin / tot_count.
+    filter_tracks: bool (optional) default: True
+        If True the track density is computed as the number of different tracks crossing a grid
+        cell. If False, the track density takes into account how long the track stayed in each
+        grid cell. Hence slower tracks increase the density if the parameter is set to False.
+
+    Returns:
+    -------
+    hist: 2D np.ndarray
+        2D matrix containing the track density
+    """
+
+    # ensure equal time step
+    # tc_track.equal_timestep(time_step_h=time_step)
```

Comment:
I think there should be a reasonable check on the track's temporal resolution versus `res` to make sure all tracks will be counted (see my comment in the other thread).

Reply:
True, there should be a limit ratio between temporal resolution and grid cell resolution. Considering that the fastest recorded translational speed of a TC is around 110 km/h, a temporal resolution of 1h and grid res of 1° (~100km) should be the limit. In this case, I would make time_step a function of res and not the opposite.